script.js

```diff
@@ -1,4 +1,159 @@
+*/
+
+//global variables for current game mode
+var playerRoll1 = 0;
+var playerRoll2 = 0;
+var gameMode = "start game by rolling 2 dice";
+var currentPlayer = 1;
+var allDiceOrder = [];
+var accumulativeRound = 0;
+
+var playerRollDice = function () {};
+
+// random dice generator for user to roll dice
+var randomDiceGenerator = function () {
+  playerRoll1 = Math.floor(Math.random() * 6 + 1);
+  playerRoll2 = Math.floor(Math.random() * 6 + 1);
+  console.log(`dice1 output:`, playerRoll1, "& dice2 output:", playerRoll2);
+
+  if (playerRoll1 == playerRoll2) {
+    return `Welcome Player ${currentPlayer}.<br> <br> You have rolled ${playerRoll1} for both dice one and dice two. <br><br>Press 1.`;
+  } else
+    return `Welcome Player ${currentPlayer}. <br> <br> You have rolled ${playerRoll1} for dice one and ${playerRoll2} for dice two.<br> <br>Choose the order of the dice by entering "1" or "2".`;
+};
+
+// function for player to roll 2 dice to get 2 number
+var chooseDiceOrder = function (userInput) {
+  orderOfDice = userInput;
+  //input validation > error message for player if they did not enter 1 or 2
+  if (userInput != 1 && userInput != 2) {
+    return "Please enter either 1 or 2 to choose which dice to put infront.";
+  }
+
+  //if user choose 1 then first dice will be infront and second dice will be behind.
+
+  if (userInput == "1") {
+    var diceOrder = Number(String(playerRoll1) + String(playerRoll2));
+    console.log(`user chose ${playerRoll1}, then ${playerRoll2}.`);
+    //game mode switches to the next player
+    gameMode = "next player roll 2 dice";
+    console.log(gameMode);
+    myOutputValue = `Player ${currentPlayer}, you have chosen dice 1. <br><br>Your final value is ${playerRoll1}${playerRoll2}.<br><br> Please click submit.`;
+  }
+
+  //if user choose 1 then first dice will be infront and second dice will be behind.
+  if (userInput == "2") {
+    diceOrder = Number(String(playerRoll2) + String(playerRoll1));
+    console.log(`user chose ${playerRoll2}, then ${playerRoll1}.`);
+    gameMode = "next player roll 2 dice";
+    //game mode switches to the next player
+    console.log(gameMode);
+    myOutputValue = `You have chosen 2. <br><br>Your final value is ${playerRoll2}${playerRoll1}. <br> <br> Please click submit.`;
+  }
+
+  //to push the user's chosen number into an array and empty the temporary variables
+
+  if (accumulativeRound == 0) {
+    allDiceOrder.push(diceOrder);
+    console.log(allDiceOrder);
+  }
+
+  if (accumulativeRound >= 1) {
+    if (currentPlayer == 1) {
+      allDiceOrder[0] = allDiceOrder[0] + diceOrder;
+      console.log(`new array value = ${allDiceOrder[0]}`);
+      diceOrder = [];
+    }
+    if (currentPlayer == 2) {
+      allDiceOrder[1] = allDiceOrder[1] + diceOrder;
+      console.log(`new array value = ${allDiceOrder[0]}`);
+      diceOrder = [];
+    }
+    playerRoll1 = [];
+    playerRoll2 = [];
+  }
+  return myOutputValue;
+};
```

Comment:
a bit unclear what `allDiceOrder` was supposed to do until I reached here and realised its an array storing the cumulative score of both players, a better name might be `cumulativeScores`? but good work in trying out the more comfortable for keeping track of the score!

also, why is `playerRoll1` and `playerRoll2` reset to an empty array, it should be 0 right? doesn't really affect your code since you assign it back to a number value during the course of the next game but just curious if this was an intentional choice or just an oversight